sdk/ai/azure-ai-client/tests/agents/test_agents_client.py

```diff
@@ -0,0 +1,1068 @@
+# # ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import os
+import json
+import time
+import functools
+import datetime
+import logging
+import sys
+
+from azure.ai.client import AzureAIClient
+from azure.ai.client.models import FunctionTool, CodeInterpreterTool, FileSearchTool, ToolSet
+from azure.core.pipeline.transport import RequestsTransport
+from devtools_testutils import AzureRecordedTestCase, EnvironmentVariableLoader, recorded_by_proxy
+from azure.core.exceptions import AzureError, ServiceRequestError, HttpResponseError
+from azure.ai.client.models import FunctionTool
+from azure.identity import DefaultAzureCredential
+
+# TODO clean this up / get rid of anything not in use
+
+'''
+issues I've noticed with the code: 
+    delete_thread(thread.id) fails
+    cancel_thread(thread.id) expires/times out occasionally
+    added time.sleep() to the beginning of my last few tests to avoid limits
+    when using the endpoint from Howie, delete_agent(agent.id) did not work but would not cause an error
+'''
+
+# Set to True to enable SDK logging
+LOGGING_ENABLED = True
+
+if LOGGING_ENABLED:
+    # Create a logger for the 'azure' SDK
+    # See https://docs.python.org/3/library/logging.html
+    logger = logging.getLogger("azure")
+    logger.setLevel(logging.DEBUG)  # INFO or DEBUG
+
+    # Configure a console output
+    handler = logging.StreamHandler(stream=sys.stdout)
+    logger.addHandler(handler)
+
+
+agentClientPreparer = functools.partial(
+    EnvironmentVariableLoader,
+    'azure_ai_client',
+    azure_ai_client_connection_string="https://foo.bar.some-domain.ms;00000000-0000-0000-0000-000000000000;rg-resour-cegr-oupfoo1;abcd-abcdabcdabcda-abcdefghijklm",
```

Comment:
Looks like we need several AI Studio hubs to run all our tests, since:
1. Evaluations at the moment run only on a private endpoint
2. You cannot have more than one AOAI connection in a hub that has an Assistant. And for endpoint tests I need multiple AOAI connections.

So my suggestion here is to have a unique connection string name for the agents tests. So instead of `azure_ai_client_connection_string` I would call it `azure_ai_client_agents_connection_string`

Reply:
Updated